### ISSUE TITLE
Bus icon should be shorter

### ISSUE THREAD

**Issue by ceraolo** [PowerGrids/PowerGrids#121]
Buses are somewhat accessory parts of diagrams.

Power system diagrams are often filled of several components, and to have clear view of everything accurate choice of model icon locations is important.

This brings to the need to put buses near to other components, e.g. generators.
An example is in the following diagram:
![image](https://github.com/PowerGrids/PowerGrids/assets/7373030/d0df5e74-19dd-4842-ad41-874bd7ea05fb)
Here I selected the bus. As you can see, the area that can be used for dragging buses (the four small red squares) overlaps generator's. 
This is suboptimal, since makes trickier to move generators on some tools: on OpenModelica, clicking on the lower part of the generator icon and dragging moves the bus instead of the generator.

I propose to make the vertical span of buses smaller. E.g. (-40,40) or (-30,30)  instead of (-100, +100).

**Comment by ceraolo** [comment]
This ticket is partly wrong. Maybe I'll reopen it in the future

**Comment by casella** [comment]
I understand the issue, and I agree that the extent should be close to the real one. However, if we do this, we affect all the existing diagrams in the library, where each and every bus will have to be resized, which is a really ugly work.

In any case, if we ever wanted to do this, we absolutely need to do it _before_ the release, otherwise we would affect all the models built with the library.

**Comment by casella** [comment]
@ceraolo, apparently if all models with diagrams are opened in OMEdit, when the extent of the icon is modified, the extents in all the diagrams that use it are updated automatically. Which means, we can do this in a very short time. @AndreaBartolini will take care of it.

**Comment by AndreaBartolini** [comment]
Unfortunately this is just an illusion created by OM... the diagrams are "graphically" updated but nothing is changed in the model annotations, so if you close and reopen OM all the diagrams have the icons enlarged and/or deformed.

We need to study another strategy...

**Comment by casella** [comment]
Wait, if the diagrams are (somehow) updated in OMEdit, it should be possible to save this information. We need to talk to @adeas31.

**Comment by casella** [comment]
The reason why the open diagrams were not changed was probably some caching. 

See OpenModelica/OpenModelica#11866 for a request of a proper solution. Let's see what @adeas31 thinks.

**Comment by casella** [comment]
BTW, I guess we should do something similar for the Synchronous Machine and Load icons. Currently, they have a lot of space in the upper half of the extent, which of course overlaps with the empty extent of the bus icon placed in the neighbourhood 😄 

This makes grabbing and moving those icons really awkward. I guess this should be fixed, if we don't want new users to get angry very soon while using the library...

**Comment by AndreaBartolini** [comment]
I've already done that for all icons in PowerGrids ;) but due to the fact that OM doesn't update the annotations of the open models :( this is the final (orrible) effect:

![immagine](https://github.com/PowerGrids/PowerGrids/assets/25974985/38a6d43e-86b6-4b0a-b30f-3c8805cd970e)

**Comment by AndreaBartolini** [comment]
obviously I've not pushed the branch....

**Comment by casella** [comment]
I'm discussing this with @adeas31. There are two steps:
- the frontend should identify which annotations must be updated
- OMEdit should compute the new extent

Not trivial, but not too difficult either.

I'm working on it.

**Comment by ceraolo** [comment]
There is another option: do not change the icon size, and let OM do like dymola, which lets grabbing icons only where there  is something (see this ticket's description).

I proposed the other way because did not require changing OM. If this, instead, is considered, before proceeding I suppose one should analyse pros and cons of the two options

**Comment by AndreaBartolini** [comment]
I agree with @ceraolo, the alternative option (grabbing icons only where there is something) applies to any library, not only to PowerGrids.

There are a lot of my libraries that suffer the same issue....

**Comment by casella** [comment]
Sounds reasonable. I'll discuss that in https://github.com/OpenModelica/OpenModelica/issues/11866

**Comment by casella** [comment]
See [comment](https://github.com/OpenModelica/OpenModelica/issues/11866#issuecomment-1910685676) in #11866.

**Comment by casella** [comment]
Nice team work 😄

**Comment by casella** [comment]
Now that OpenModelica/OpenModelica#11866 is fixed, also OpenModelica allows to grab only the drawn portion of the extent, so there is no need to actually change it.
